node/full.go

```diff
@@ -1,39 +1,32 @@
 package node
 
 import (
-	"context"
-
 	"go.uber.org/fx"
 
-	coreclient "github.com/celestiaorg/celestia-node/core"
-	"github.com/celestiaorg/celestia-node/node/core"
-	"github.com/celestiaorg/celestia-node/node/p2p"
+	"github.com/celestiaorg/celestia-node/core"
+	"github.com/celestiaorg/celestia-node/node/config"
+	nodecore "github.com/celestiaorg/celestia-node/node/core"
+	"github.com/celestiaorg/celestia-node/node/fxutil"
 	"github.com/celestiaorg/celestia-node/service/block"
 )
 
 // NewFull assembles a new Full Node from required components.
-func NewFull(cfg *Config) (*Node, error) {
-	return newNode(fullComponents(cfg))
+func NewFull(cfg *config.Config, corecfg *core.Config) (*Node, error) {
```

Comment:
Agh i don't really like this cause we have a goal to also run celestia full node standalone (without a trusted core process) - can `core config` be part of the overall `Config`?